Library/Homebrew/system_command.rb

```diff
@@ -54,15 +54,17 @@ def run!
   end
 
   def initialize(executable, args: [], sudo: false, env: {}, input: [], must_succeed: false,
-                 print_stdout: false, print_stderr: true, verbose: false, **options)
+                 print_stdout: false, print_stderr: true, verbose: false, secrets: [], **options)
 
+    require "extend/ENV"
     @executable = executable
     @args = args
     @sudo = sudo
     @input = [*input]
     @print_stdout = print_stdout
     @print_stderr = print_stderr
     @verbose = verbose
+    @secrets = Array(secrets) + ENV.sensitive_environment.map(&:last)
```

Comment:
`ENV.sensitive_environment.values` and maybe worth having a `.uniq` for the whole thing?